File: URich/app/src/main/java/com/example/urich/MainActivity.java
package com.example.urich;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.CheckBox;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    boolean emma;
    TextView numberView;
    Random random=new Random();
    int getal=0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        numberView=findViewById(R.id.numberView);
        emma=false;
    }

    public void Change(View view) {
        if(emma){
            emma=false;
        }else{
            emma=true;
        }
    }

    public void Roll(View view) {
        if(emma){
            getal = random.nextInt(7-4)+4;
            Log.d("getal","EMMA"+getal);
        }else {
            getal = random.nextInt(7-1)+1;
            Log.d("getal",""+getal);
        }
        numberView.setText(String.valueOf(getal));
    }

    public void Reset(View view) {
        numberView.setText("");
    }
}

File: DiceRoller/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import java.util.*

class MainActivity : AppCompatActivity() {

    lateinit var diceImage:ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        diceImage=findViewById(R.id.dice_image)
        val rollButton: Button = findViewById(R.id.roll_button)
        rollButton.setOnClickListener { rollDice() }
/*        val countButton: Button = findViewById(R.id.countup_button)
        countButton.setOnClickListener { oneUp() }*/
        val resetButton: Button= findViewById(R.id.reset_button)
        resetButton.setOnClickListener{reset()}
    }
    private fun rollDice(){
        val randomget = Random().nextInt(6)
        val drawableImage = when (randomget) {
            1->R.drawable.dice_1
            2->R.drawable.dice_2
            3->R.drawable.dice_3
            4->R.drawable.dice_4
            5->R.drawable.dice_5
            else->R.drawable.dice_6
        }
        diceImage.setImageResource(drawableImage)
    }
    /*private fun oneUp(){
        val diceImage: ImageView = findViewById(R.id.dice_image)
         if (resultText.text == "Hello World!") {
           resultText.text = "1"
        } else {
            var resultInt = resultText.text.toString().toInt()
            if (resultInt < 6) {
                resultInt++
                resultText.text = resultInt.toString()
            }
        }
    }*/
    private fun reset(){
        diceImage.setImageResource(R.drawable.empty_dice)
    }
}
